src/Microsoft.AspNetCore.Rewrite/Internal/IISUrlRewrite/IISUrlRewriteRule.cs

```diff
@@ -57,6 +65,22 @@ public virtual void ApplyRule(RewriteContext context)
                 }
             }
 
+            foreach (ServerVariable serverVariable in ServerVariables)
+            {
+                if (serverVariable.Name.StartsWith(RequestHeaderPrefix, StringComparison.CurrentCultureIgnoreCase))
+                {
+                    context.HttpContext.Request.Headers.Add(
+                        serverVariable.Name.Substring(RequestHeaderPrefix.Length).Replace('_', '-'),
+                        serverVariable.Evaluate(context, initMatchResults, condMatchRes));
+                }
+                else if (serverVariable.Name.StartsWith(ResponseHeaderPrefix, StringComparison.CurrentCultureIgnoreCase))
+                {
+                    context.HttpContext.Response.Headers.Add(
+                        serverVariable.Name.Substring(ResponseHeaderPrefix.Length).Replace('_', '-'),
+                        serverVariable.Evaluate(context, initMatchResults, condMatchRes));
```

Comment:
Would be great if we can get some logging in here when we we add a server variable (less of a priority).